From: Mort Yao <soi@mort.ninja>
Date: Fri, 10 Aug 2018 05:54:00 +0100
Subject: [PATCH] Legacy tool order

---

--- a/app/tools/gimp-tools.c	2018-07-04 13:00:26.000000000 +0200
+++ b/app/tools/gimp-tools.c	2018-08-10 05:46:43.853140768 +0200
@@ -144,15 +144,15 @@
     gimp_move_tool_register,
     gimp_align_tool_register,
     gimp_crop_tool_register,
-    gimp_unified_transform_tool_register,
+    // gimp_unified_transform_tool_register,
     gimp_rotate_tool_register,
     gimp_scale_tool_register,
     gimp_shear_tool_register,
-    gimp_handle_transform_tool_register,
+    // gimp_handle_transform_tool_register,
     gimp_perspective_tool_register,
     gimp_flip_tool_register,
     gimp_cage_tool_register,
-    gimp_warp_tool_register,
-    gimp_n_point_deformation_tool_register,
+    // gimp_warp_tool_register,
+    // gimp_n_point_deformation_tool_register,

     /*  paint tools  */

-    gimp_seamless_clone_tool_register,
+    // gimp_seamless_clone_tool_register,
     gimp_text_tool_register,
     gimp_bucket_fill_tool_register,
     gimp_gradient_tool_register,
@@ -166,7 +166,7 @@
     gimp_eraser_tool_register,
     gimp_airbrush_tool_register,
     gimp_ink_tool_register,
-    gimp_mybrush_tool_register,
+    // gimp_mybrush_tool_register,
     gimp_clone_tool_register,
     gimp_heal_tool_register,
     gimp_perspective_clone_tool_register,
