
--- gnome-shell/js/ui/panel.js	2019-10-26 23:22:42.617368518 +0200
+++ gnome-shell/js/ui/panel.js	2019-10-26 23:22:01.397370298 +0200
@@ -218,15 +218,15 @@
         textureCache.connect('icon-theme-changed',
                              this._onIconThemeChanged.bind(this));

-        let iconEffect = new Clutter.DesaturateEffect();
+        //let iconEffect = new Clutter.DesaturateEffect();
         this._iconBox = new St.Bin({ style_class: 'app-menu-icon' });
-        this._iconBox.add_effect(iconEffect);
+        //this._iconBox.add_effect(iconEffect);
         this._container.add_actor(this._iconBox);

-        this._iconBox.connect('style-changed', () => {
-            let themeNode = this._iconBox.get_theme_node();
-            iconEffect.enabled = themeNode.get_icon_style() == St.IconStyle.SYMBOLIC;
-        });
+        //this._iconBox.connect('style-changed', () => {
+        //    let themeNode = this._iconBox.get_theme_node();
+        //    iconEffect.enabled = themeNode.get_icon_style() == St.IconStyle.SYMBOLIC;
+        //});

         this._label = new St.Label({ y_expand: true,
                                      y_align: Clutter.ActorAlign.CENTER });
